packages/puppeteer-core/src/common/types.ts

```diff
@@ -9,6 +9,11 @@ import type {JSHandle} from '../api/JSHandle.js';
 
 import type {LazyArg} from './LazyArg.js';
 
+/**
+ * @public
+ */
+export type AwaitablePredicate<T> = (value: T) => Awaitable<boolean>;
```

Comment:
Q: Didn't you add it something similar in another PR?

Reply:
In the original bidi/core PR I made this refactor. Separated it out.